python/lsst/pipe/tasks/multiBand.py

```diff
@@ -834,8 +990,20 @@ def runDataRef(self, patchRef, psfCache=100):
         sources = self.readSources(patchRef)
         table = sources.getTable()
         table.setMetadata(self.algMetadata)  # Capture algorithm metadata to write out to the source catalog.
+        skyInfo = getSkyInfo(coaddName=self.config.coaddName, patchRef=patchRef)
+
+        results = self.run(exposure=exposure, sources=sources,
+                           ccdInputs=self.propagateFlags.getCcdInputs(exposure),
+                           skyInfo=skyInfo, butler=patchRef.getButler(),
+                           exposureId=self.getExposureId(patchRef))
+
+        if self.config.doMatchSources:
+            self.writeMatches(patchRef, results)
+        self.write(patchRef, results.outputSources)
 
-        self.measurement.run(sources, exposure, exposureId=self.getExposureId(patchRef))
+    def run(self, exposure, sources, skyInfo, exposureId, ccdInputs=None, visitCatalogs=None, wcsUpdates=None,
+            butler=None):
```

Comment:
I know we're just trying to get things done quickly right now and plan to clean up in the future, but adding parameter docs for this method might be worth doing on this ticket, especially for anyone who might need to make changes to the Gen2 version before we retire it.